doc/src/devdocs/ast.md

```diff
@@ -757,3 +763,12 @@ Boolean properties:
 
     Whether this is known to be a pure function of its arguments, without respect to the
     state of the method caches or other mutable global state.
+
+
+`UInt8` settings:
+
+  * `constprop`
```

Comment:
We also want to include the `purity` setting now here also (it can be added in another PR).